sphinx_gallery/gen_gallery.py

```diff
@@ -687,6 +692,150 @@ def write_computation_times(gallery_conf, target_dir, costs):
             fid.write(hline)
 
 
+def write_api_entry_usage(gallery_conf, target_dir):
+    if gallery_conf['backreferences_dir'] is None:
+        return
+
+    backreferences_dir = os.path.join(gallery_conf['src_dir'],
+                                      gallery_conf['backreferences_dir'])
+    example_files = [example for example in os.listdir(backreferences_dir)
+                     if '__' not in example and
+                     ((example.endswith('.examples') and
+                       not os.path.isfile(example + '.new')) or
+                     example.endswith('.examples.new'))]
```

Comment:
I think this is much more readable, but YMMV. If you go with my suggestion please double-check I captured the logic correctly:
```suggestion

    def func(fname):
        is_new = fname.endswith('.examples.new')
        is_old = os.path.isfile(fname + '.new')
        is_ex = fname.endswith('.examples')
        is_builtin = '__' in fname
        return not is_builtin and is_new or (is_ex and not is_old)

    example_files = list(filter(func, os.listdir(backreferences_dir)))
```